torch/csrc/StorageMethods.cpp

```diff
@@ -173,8 +173,8 @@ static PyObject* THPStorage_fromBuffer(
   PyObject* dtype_obj = nullptr;
   c10::ScalarType scalar_type = at::kByte;
   Py_buffer buffer = {};
-  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,clang-diagnostic-writable-strings)
-  static char* kwlist[] = {
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  constexpr const char* kwlist[] = {
```

Comment:
Don't think just need constexpr or const, not both. Not sure if that solves the build error though.

Reply:
I am afraid that discard const would trigger the warnings, so I decide to cast kwlist

Reply:
@cyyever Actually, I am 90% sure it won't trigger the warnings. THe casting is necessary because it's interpreting it as a `const char* const` which is not the same as a const char*. Just remove the second const and the warning should disappear. The alternative casting is an unsafe const cast which violates the whole point of the warning in the first place.

Reply:
To be honest, I think the CPython API should use const char* const char* as the type of kwlist, I am testing constexpr solely in my local env now. If it works, I would push another commit.

Reply:
@cyyever From my experience in pybind11, I think that's only the most recent versions of Python.